lineapy/plugins/pipeline_writers.py

```diff
@@ -0,0 +1,206 @@
+import logging
+from pathlib import Path
+from typing import Dict, List, Optional
+
+from lineapy.api.api_utils import de_lineate_code
+from lineapy.graph_reader.node_collection import NodeCollectionType
+from lineapy.graph_reader.session_artifacts import SessionArtifacts
+from lineapy.plugins.task import AirflowDagConfig, AirflowDagFlavor, TaskGraph
+from lineapy.plugins.utils import load_plugin_template
+from lineapy.utils.logging_config import configure_logging
+from lineapy.utils.utils import prettify
+
+logger = logging.getLogger(__name__)
+configure_logging()
+
+
+class BasePipelineWriter:
+    """
+    Base class for pipeline file writer. Corresponds to "SCRIPT" framework.
+    """
+
+    def __init__(
+        self,
+        session_artifacts_sorted: List[SessionArtifacts],
+        keep_lineapy_save: bool,
+        pipeline_name: str,
+        output_dir: str,
+    ) -> None:
+        self.session_artifacts_sorted = session_artifacts_sorted
+        self.keep_lineapy_save = keep_lineapy_save
+        self.pipeline_name = pipeline_name
+        self.output_dir = Path(output_dir, pipeline_name)
+
+        # Create output directory folder(s) if nonexistent
+        self.output_dir.mkdir(exist_ok=True, parents=True)
+
+        # We assume there is at least one SessionArtifacts object
+        self.db = self.session_artifacts_sorted[0].db
+
+    def _write_dag(self) -> None:
```

Comment:
Can we get this to have the same API as the _write_dag for Airflow Pipeline writer? 

This might be a bit difficult because you'll probably need to do something like 
def _write_dag(self, dag_config: Optional[BaseDagConfig]) -> None:

where BaseDagConfig is just empty and unused, and then get AirflowDagConfig to inherit BaseDagConfig. 

This way the two classes actually inherit each other and have the same interface.

Reply:
Good suggestion, working on it